controller/volume_controller.go

```diff
@@ -1005,15 +1013,19 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		}
 	}
 
-	if e.Spec.LogRequested && e.Status.LogFetched {
-		e.Spec.LogRequested = false
-	}
+	allReplicasStopRequestingLog := true
```

Comment:
NIT: I prefer the inverse, but I don't have strong feelings about it, you can also keep the if if you prefer.
```
needReplicaLogs := false
needReplicaLogs = needReplicaLogs || r.spec.LogRequested
needEngineLogs := needReplicaLogs || (e.spec.LogRequested && !e.spec.LogFetched)
e.spec.LogRequested = needEngineLogs
```

Reply:
Fixed to use the inverse. Not identical to the suggested change since we only reset the value here. Please review